docs/advanced/payments/existing-domain-events.mdx

```diff
@@ -0,0 +1,143 @@
+---
+sidebar_position: 5.4
+title: Existing Domain Events
+description: See the detailed list of existing Domain Events.
+---
+
+# Payment events
+
+Payment events allow to inform payment event listeners about a payment status
+change.
+
+Event listeners can use this information to maintain a consistent state in a
+remote system. For instance, an eCommerce integration will implement an Order
+event listeners responsible for updating the order status accordingly on a
+headless eCommerce platform.
+
+Note: each event is based on a `PurchaseIdentifier` allowing Front-Commerce to
+retrieve the related order based on either an orderId or the cartId that was
+used before the order creation.
+
+## The paymentData attribute
+
+Most events allows you to provide custom, optional and arbitrary `paymentData`
+values. These values may contain data specific to the payment platform. We
+recommend that event listeners push these information to remote systems as they
+usually can be helpful for further processing (e.g: synchronizing a headless
+eCommerce platform with an ERP)
+
+Example: Front-Commerce Magento's integration, will append this `paymentData` as
+a JSON payload to the order history comment message.
+
+## PaymentAuthenticationStarted
+
+The authentication of the payment started (e.g. 3D Secure control step)
+
+```js
+import PaymentAuthenticationStarted from "server/modules/front-commerce/payment/domain/events/PaymentCaptureStarted";
+
+return [new PaymentAuthenticationStarted(new PurchaseIdentifier({ orderId }, pspReference, authenticationMethodName, paymentData)),
```

Comment:
I guess it should be:

```suggestion
return [new PaymentAuthenticationStarted(new PurchaseIdentifier({ orderId }), pspReference, authenticationMethodName, paymentData))];
```

but why returning an array ? and even why do we even put that code in the documentation ? I mean the class is already properly annotated so this does bring anything except that we'll have to maintain both the code and the example here.

Instead, I think this page should just list all events with a description of what they means. The parameter description should be in the doc block (if needed)

Reply:
In fact, integrators are copy/pasting the code from the documentation, checking the implementation is an effort they only bear when they have no other choice. 

(I do the same most of the time : when I don't know what to write, I struggle more than get the idea to search for the doc in the node_modules)

Reply:
if you put a link to the code, the instantiating the event is trivial and we avoid the risk of having an inconsistent documentation.

Reply:
You're right, I put links in the doc and improved the jsdoc, see : https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1506